crypto/Kconfig

```diff
@@ -39,6 +39,11 @@ config CRYPTO_CRYPTODEV
 	bool "cryptodev support"
 	default n
 
+config CRYPTO_CRYPTODEV_HARDWARE
+	bool "cryptodev hardware support"
+	depends on CRYPTO_CRYPTODEV
+	default n
+
 config CRYPTO_SW_AES
```

Comment:
`CRYPTO_SW_AES`  only use is being removed in this PR, so this config should be either removed or reworked.

I believe it should be renamed to something like `CRYPTO_SW_IMPL` to consider the newly supported algorithms. Besides, the `depends on ALLOW_BSD_COMPONENTS` is still valid for AES and the others, right?

Reply:
yes

Reply:
I am referring to just the Kconfig options, so why not do it on this PR?

PRs should bring self-contained changes. If I understood correctly, this PR in its current state is adding new BSD-licensed components, so the correct would be properly "hide" them behind the `ALLOW_BSD_COMPONENTS` config.

Reply:
ok i add `ALLOW_BSD_COMPONENTS` to `CRYPTO_CRYPTODEV`